scrapy_poet/downloadermiddlewares.py

```diff
@@ -148,6 +148,11 @@ def process_response(
         )
         # Fill the callback arguments with the created instances
         for arg, value in final_kwargs.items():
+            # If scrapy-poet can't provided the dependency, allow the user to
+            # give it.
+            if value is None and request.cb_kwargs.get(arg):
```

Comment:
It seems we can replace user-provided False or 0 with None, because of the way condition is written.

Reply:
Good catch! https://github.com/scrapinghub/scrapy-poet/pull/113/commits/800d512f97d0e9a71df0f0dd07d480b674beaa39